### PR TITLE
Fixed corp num not being populated in CHG, REH, and CNV

### PR DESCRIPTION
**Pull request by JazzarKarim**
*Issue #:* /bcgov/entity#17633

*Description of changes:*
- The `corpNum` field is now being populated with the business identifier after submitting a CHG, REH, or CNV
- An example after submitting a Restoration:
corp num now works

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the namerequest license (Apache 2.0).

### CODE REVIEW

File: src/store/actions.ts
Comment (JazzarKarim) [inline src/store/actions.ts:167]: After doing lots of digging, I found out that every time, the `corpNum` is being set as an empty string here. There was nothing to counter this for Alterations, Restorations, and Name Change.

So, I moved this line to the else statement below.

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://namerequest-dev--pr-676-mfoo44bx.web.app

File: src/store/actions.ts
Comment (severinbeauvais) [inline src/store/actions.ts:271]: I don't think the purpose of this action ("commitExistingData") is to clear fields in the newRequestModel object, so I don't understand why you're doing this here.

File: src/store/actions.ts
Comment (JazzarKarim) [reply src/store/actions.ts:167]: Sev, please check latest commit.

Here's my thought process: The `corpNum` is being manually set in the code in a couple of places. When it's been manually set (like in REH for example), I don't want to run mutate the `corpNum` to being an empty string.

Comment (severinbeauvais) [reply src/store/actions.ts:167]: This "feels" better. It's clearing the value in the same place as the others, and there's similar code to yours just above. So it's probably OK.

If we had time, it would be nice to understand why `resetAnalyzeName` is being called after we set a value. But we don't, so if this works then I'm fine with it.

Comment (severinbeauvais) [reply src/store/actions.ts:167]: How about adding a comment that we set corpNum earlier in the path and we don't want to reset it here while analyzing the name?

(Now it "feels" like corpNum has nothing to do with name analysis, so why is it being cleared here at all? Sigh.)

**Review by severinbeauvais — APPROVED** [review]

File: src/store/actions.ts
Comment (severinbeauvais) [reply src/store/actions.ts]: See comments above. Resolving.

**Comment by severinbeauvais** [comment]
Karim, I think you need to rebase and update the app version, and then you can merge this.

File: src/store/actions.ts
Comment (JazzarKarim) [reply src/store/actions.ts:167]: You mean add a comment in the code explaining why is the if statement there Sev and why are we setting the corpNum as an empty string? I didn't quite get that.

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://namerequest-dev--pr-676-mfoo44bx.web.app

File: src/store/actions.ts
Comment (severinbeauvais) [reply src/store/actions.ts:167]: Yes, why is this here / needed? But this may all be obsolete if you remove corpNum from the name analysis flow (since it's not at all related, I think).

The root cause is that we previously did name analysis and then prompted for the corpNum, but now we look up the corpNum and then do the name analysis.

Comment (JazzarKarim) [reply src/store/actions.ts:167]: Yes, agreed. This line was written 2 years ago. I think it is indeed obsolete. Going to take a deeper dive and make sure if it's obsolete, I can remove the line all together.

Comment (JazzarKarim) [reply src/store/actions.ts:167]: I've confirmed Sev. This line is indeed obsolete.

I've tried going through the process of creating an NR through various flows after removing this line. The corpNum is being properly set when REH is selected for example while it's empty in a NEW flow.

Comment (severinbeauvais) [reply src/store/actions.ts:167]: Thanks for testing. Are the other 2 flows OK? And so this PR is not needed?

Comment (severinbeauvais) [reply src/store/actions.ts:167]: Awesome. This conversation is fine by me (ok to resolve).

Comment (severinbeauvais) [reply src/store/actions.ts:167]: Why don't you merge this fix and open another PR for other fixes?

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://namerequest-dev--pr-676-mfoo44bx.web.app

File: src/store/actions.ts
Comment (JazzarKarim) [reply src/store/actions.ts:167]: Sounds good. I'm just waiting for another approval!

**Review by ketaki-deodhar — APPROVED** [review]
LGTM!

**Pull request merged**
